File: stacks/laravel/tests/Feature/HealthTest.php
<?php

namespace Tests\Feature;

use Tests\TestCase;

/**
 * 死活監視。契約は /healthz（spec/openapi.yaml）。
 *
 * ここが緑でも「終わった」ではない。完了の判定は共通テスト
 * （python tests/run.py <URL>）が緑になったときだけ。
 */
class HealthTest extends TestCase
{
    public function test_healthz_returns_ok(): void
    {
        $response = $this->get('/healthz');

        $response->assertOk();
        $response->assertHeader('Content-Type', 'application/json');
        $response->assertExactJson(['status' => 'ok']);
    }

    public function test_health_alias_still_answers(): void
    {
        $this->get('/health')->assertOk()->assertExactJson(['status' => 'ok']);
    }
}

File: stacks/rails/test/integration/health_test.rb
require "test_helper"

class HealthTest < ActionDispatch::IntegrationTest
  # 共通テスト（tests/run.py の smoke）が叩くのは /healthz。
  # 起動文面には /health と書いてあったので、両方に応える。
  [ "/healthz", "/health" ].each do |path|
    test "GET #{path} は {\"status\":\"ok\"} を返す" do
      get path

      assert_response :success
      assert_equal({ "status" => "ok" }, JSON.parse(response.body))
    end
  end

  # 共通テストのクライアントは urllib で、この Accept と UA を送ってくる。
  # 画面向けの allow_browser / レイアウトに巻き込まれて 406 にならないことを確かめる。
  test "共通テストと同じ Accept・User-Agent でも 200 を返す" do
    get "/healthz", headers: {
      "Accept" => "text/html,application/json",
      "User-Agent" => "Python-urllib/3.14"
    }

    assert_response :success
    assert_equal({ "status" => "ok" }, JSON.parse(response.body))
  end
end
